### PR TITLE
Update pods and fix compile errors

### PR DESCRIPTION
**Pull request by muinmomin**
- Updated pods and fixed compile errors
- Quick validation that app launches in simulator, able to sign-in, all screens load

### CODE REVIEW

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 1]
```diff
@@ -277,8 +278,28 @@ extension LoginV2ViewController {
   
   @objc
   private func googleLoginButtonTapped() {
-    GIDSignIn.sharedInstance()?.presentingViewController = self
-    GIDSignIn.sharedInstance()?.signIn()
+    let config = GIDConfiguration(clientID: FirebaseApp.app()?.options.clientID ?? "")
```
Comment (muinmomin) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]: `clientId ?? ""` -> should this be an error case instead?

Comment (compnerd) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: Yes, I think that should be an error scenario.

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 2]
```diff
@@ -277,8 +278,28 @@ extension LoginV2ViewController {
...
+      }
+      
+      guard let authentication = user?.authentication else {
+        // This shouldn't happen if error == nil and we don't have a way to recover except try again
+        NotificationCenter.default.post(name: .googleSigninFailure, object: "Unknown Error")
+        return
+      }
+      
+      guard let idToken = authentication.idToken else {
+        NotificationCenter.default.post(name: .googleSigninFailure, object: "Unknown Error")
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]: Should we differentiate between L290 and L295?

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 3]
```diff
@@ -330,3 +351,9 @@ extension LoginV2ViewController: ASAuthorizationControllerDelegate, ASAuthorizat
     AppController.shared.present(alert: alert, in: self, completion: nil)
   }
 }
+
+
+extension Foundation.Notification.Name {
+  static let googleSignInSuccess = Foundation.Notification.Name(rawValue: "googleSignInSuccess")
+  static let googleSigninFailure = Foundation.Notification.Name(rawValue: "googleSigninFailure")
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]: I think that we should be explicit here about the extension being private.  Extensions on unowned types is something to be avoided.

```suggestion
  internal static let googleSignInSuccess = Foundation.Notification.Name(rawValue: "googleSignInSuccess")
  internal static let googleSigninFailure = Foundation.Notification.Name(rawValue: "googleSigninFailure")
  internal static let firebaseFailure = Foundation.Notification.Name(rawValue: "firebaseFailure")
```

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 1 above)
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]:
```suggestion
    guard let clientID = FirebaseApp.app()?.options.clientID else {
      NotificationCenter.default.post(name: .firebaseFailure, object: "missing client ID")
      return
    }
    let config = GIDConfiguration(clientID: clientID)
```

**Comment by compnerd** [comment]
@muinmomin any idea about the test failure: https://dev.azure.com/steps4impact/steps4impact/_build/results?buildId=741&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=e93be99f-5c53-5999-419d-b30edfa9bdfd&l=2132

I had assumed that this was due to a pod version conflict.  https://github.com/AgaKhanFoundation/WCF-iOS/pull/422 was a previous hack that I had applied to work around it.  I'd prefer if we could understand what is going on, but I suppose that we could always re-apply the hack :-(.

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 1 above)
Comment (muinmomin) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: Updated to be an error

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 2 above)
Comment (muinmomin) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: Wasn't able to find a user facing error message for missing idToken. I'm assuming that either both authentication and idToken will be missing, or neither. Updated to (per: https://firebase.google.com/docs/auth/ios/google-signin):
```swift
guard
        let authentication = user?.authentication,
        let idToken = authentication.idToken
      else {
```

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 3 above)
Comment (muinmomin) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: Thanks, good to know, updated!

**Review by compnerd — APPROVED** [review]

**Pull request merged**